Mathlib/Geometry/Manifold/VectorField.lean

```diff
@@ -0,0 +1,339 @@
+/-
+Copyright (c) 2024 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import Mathlib.Analysis.Calculus.VectorField
+import Mathlib.Geometry.Manifold.ContMDiffMFDeriv
+import Mathlib.Geometry.Manifold.MFDeriv.NormedSpace
+import Mathlib.Geometry.Manifold.VectorBundle.MDifferentiable
+
+/-!
+# Vector fields in manifolds
+
+We study functions of the form `V : Π (x : M) → TangentSpace I x` on a manifold, i.e.,
+vector fields.
+
+We define the pullback of a vector field under a map, as
+`VectorField.mpullback I I' f V x := (mfderiv I I' f x).inverse (V (f x))`
+(together with the same notion within a set).
+
```

Comment:
Given that this is a wrong-way map, I think we should highlight how it can exist. Maybe something like:
```suggestion
Note that the pullback uses the junk-value pattern: if the derivative of the map is not invertible,
then pullback is given the junk value zero.
```
(I appreciate this is already flagged one level down in the documentation for affine story, but I think it deserves repeating here.)